components/new-contribution-flow/StepDetails.js

```diff
@@ -64,7 +64,7 @@ const StepDetails = ({ onChange, data, collective, tier, showFeesOnTop, isTaxDed
           </StyledInputField>
         </Box>
       )}
-      {(!tier || tier.amountType === 'FLEXIBLE') && (
+      {(!tier || tier?.amountType === 'FLEXIBLE') && (
```

Comment:
This one should not be necessary, if the tier doesn't exist it will match the first condition (`!tier`)

```suggestion
      {(!tier || tier.amountType === 'FLEXIBLE') && (
```